src/Ioc/index.js

```diff
@@ -133,11 +133,15 @@ class Ioc {
      * @param parameters
      * @return {*}
      */
-    static call(binding, functionName, parameters = []) {
+    static async call(binding, functionName, parameters = []) {
```

Comment:
call function cannot be async, as it's used by several different things, a lot of stuff would change if we make it async.

Reply:
Async/await can be used without returning promises.
A sentence like:
```js
const result = await someFunction();
```
Assigns value to the result even if `someFunction` doesn't return a promise.
Same as:
```js
const result = someFunction();
```

So, current proposal won't change anything, as `call` function can be used both with and without `async`.

Reply:
All the tests needed are added to the respected test files.